### PR TITLE
audio/engine.asm and home/audio.asm restyling

### PR DESCRIPTION
**Pull request by aljohnston112**
I would like some feedback on my restyling before continuing. Are there things I'm doing wrong or could be done another way?

### CODE REVIEW

File: audio/engine.asm [hunk 1]
```diff
@@ -1197,12 +1216,15 @@ ParseMusic:
...
 .sound_ret
-; $ff is reached in music data
 	ld hl, CHANNEL_FLAGS1
 	add hl, bc
-	bit SOUND_SUBROUTINE, [hl] ; in a subroutine?
-	jr nz, .readcommand ; execute
+	; in a subroutine?
+	bit SOUND_SUBROUTINE, [hl] 
+	; execute
+	jr nz, .readcommand
```
Comment (aaaaaa123456789) [inline audio/engine.asm:1227]: If these comments weren't redundant, which they are, they should remain where they are. But the main problem is that the comments aren't needed in the first place — it should be obvious that `bit SOUND_SUBROUTINE, [hl]` is checking whether the engine is in a subroutine. These comments might've been necessary back when these were unnamed constants or simple numbers, but right now, that instruction is self-documenting, and so is the jump.

File: audio/engine.asm [hunk 2]
```diff
@@ -1356,6 +1379,7 @@ ParseMusicCommand:
 	ld d, 0
 	; seek command pointer
 	ld hl, MusicCommands
+	; each command takes up two bytes
```
Comment (aaaaaa123456789) [inline audio/engine.asm:1382]: This is pretty much implied by the fact that the command offset is doubled before being added to the pointer. Labelling all such indexing operations with a comment would create lots of noise; it's a very common pattern.

File: audio/engine.asm [hunk 3]
```diff
@@ -1446,9 +1470,14 @@ Music_Ret:
...
 
-Music_Call:
 ; call music stream (subroutine)
-; parameters: ll hh ; pointer to subroutine
+; input: 
+	; bc = start of current channel
+; output: 
+	; CHANNEL_LAST_MUSIC_ADDRESS = CHANNEL_MUSIC_ADDRESS
+	; CHANNEL_MUSIC_ADDRESS = pointer to subroutine (ll hh)
+	; Sets the SOUND_SUBROUTINE flag in CHANNEL_FLAGS1
```
Comment (aaaaaa123456789) [inline audio/engine.asm:1479]: `CHANNEL_LAST_MUSIC_ADDRESS` and `CHANNEL_MUSIC_ADDRESS` are constants, and thus, they cannot ever be outputs in the first place. In fact, all inputs and outputs must be in a register or a memory location, and those are neither. Are you sure you're understanding the function in the first place?

Also, this excess verbosity in function comments isn't helpful; we're not trying to run doxygen on the codebase. The files are already massive as it is; keep your comments succint and to the point.

File: audio/engine.asm [hunk 4]
```diff
@@ -1479,9 +1508,13 @@ Music_Call:
...
 	ret
 
-Music_Jump:
 ; jump
-; parameters: ll hh ; pointer
+; input: 
+	; bc = start of current channel
+; output: 
+	; CHANNEL_MUSIC_ADDRESS = pointer to 
+	; music address to jump to (ll hh)
```
Comment (aaaaaa123456789) [inline audio/engine.asm:1516]: Again, this excess verbosity isn't helpful. Keep it all in one line. (And again, `CHANNEL_MUSIC_ADDRESS` cannot be an output. Are you sure you're understanding what's going on?)

The important part here, after all, is that the two bytes that follow the jump command itself in the channel stream represent the jump target address, in little-endian. The original comment already conveyed that information. The rest can easily be left out.

File: audio/engine.asm [hunk 5]
```diff
@@ -2101,10 +2134,13 @@ Music_NewSong:
...
 	ret
 
-GetMusicByte:
 ; returns byte from current address in a
 ; advances to next byte in music data
-; input: bc = start of current channel
+; input: 
+	; bc = start of current channel
+; output: 
+	; a = music byte
```
Comment (aaaaaa123456789) [inline audio/engine.asm:2142]: Once again, cut down the verbosity. The function comment already said that the function returned in `a`, so there's no point in saying that twice.

File: audio/engine.asm [hunk 6]
```diff
@@ -2117,19 +2153,18 @@ GetMusicByte:
 	ld hl, CHANNEL_MUSIC_BANK
 	add hl, bc
 	ld a, [hl]
-	; get byte
-	call _LoadMusicByte ; load data into wCurMusicByte
-	inc de ; advance to next byte for next time this is called
+	; load byte into wCurMusicByte
+	call _LoadMusicByte
+	; advance to next byte for next time this is called and
```
Comment (aaaaaa123456789) [inline audio/engine.asm:2158]: By moving this comment outside of the line where it was, you have made the flow more obscure. Before, it was clear that `inc de` was simply incrementing the pointer for the next time, and that the following code was updating the channel data address. Now that's merged into a single comment, still broken down into two lines for some unexplainable reason.

File: home/audio.asm [hunk 7]
```diff
@@ -48,8 +48,12 @@ UpdateSound::
 	pop hl
 	ret
 
+; input: 
+	; a = channel music bank
+	; de = channel music address
+; output:
+	; wCurMusicByte = [a:de]
```
Comment (aaaaaa123456789) [inline home/audio.asm:55]: This is all easily implied by the single comment that was already there. What's the benefit in this over-verbosity?

**Review by aaaaaa123456789 — CHANGES_REQUESTED** [review]
Honestly, most of these changes seem like they'd be doing a net damage to the codebase. I see a lot of changes "just because Visual Studio says so" (comments above function definitions instead of inside, pointlessly verbose calling convention descriptions, and so on) and many changes made by blindly applying rules without considering why things were done the way they were. It's a shame, because `audio/engine.asm` is a very old file that could definitely use some love, but this isn't helping.

To maintainers: maybe the style guide needs an update? There are many places in which things deviate from that guide, and while they probably seem very intuitive for us, we're not the ones checking the guide in the first place. If this is the kind of thing that the style guide leads to, I'd rather see the file wholly deleted.

**Comment by Rangi42** [comment]
Thank you for your interest in improving pokecrystal! However, this PR is not a great start to that.

I see ax6 is going over some particular lines, but my overall comments:

- IDE-specific files like `.vs/*` don't belong in `.gitignore`
- Don't indent comments by two spaces
- No need to be absolutely consistent about whether comments go above vs below a label, or above vs next to an opcode
- Some comments like "`store next byte in a`" for `call GetMusicByte`, or "`load byte into wCurMusicByte`" for `call _LoadMusicByte`, are just noise (note that some files disassembled early on, like the audio engine and much of pokered, have excessive comments already, since fewer labels and constants existed at the time)

**Comment by Rangi42** [comment]
> To maintainers: maybe the style guide needs an update? There are many places in which things deviate from that guide, and while they probably seem very intuitive for us, we're not the ones checking the guide in the first place.

Yes, it should be revised. For example, short comments next to short code lines are okay; and there's no obvious best choice for comments going above vs below a function label, or comments above "paragraphs" of code being indented vs not, so those are  probably better off left to individual judgment than trying to standardize the whole codebase.

**Pull request closed without merging**